### ISSUE TITLE
Playback does not start with ALSA softvol OR dmix devices

### ISSUE THREAD

**Issue by Foldi-One** [popcornmix/omxplayer#718]
I defined a dmix and softvol device in ~/asoundrc for a cirrus audio card. (It has no master vol for its many outputs and I want to dmix so MOCP can be left running)
Calling omxplayer -o:alsa or -o:alsa:"alias of dmix/softvol devices"  there are no errors, but CPU load drops to idle and playback never starts. Works just fine using alias of cirrus card. (and the alsa devices work just fine)

**Comment by matthijskooijman** [comment]
I think this might be related to #745.

Do you see any errors in the generated logfile if you run with `--genlog`?

**Comment by Foldi-One** [comment]
Can't reproduce this anymore seems a some update or other has fixed whatever was causing it.. As far as I recall nothing in the config has changed but I tried it again and it just worked this time.
Unlike your error #745  which gives a line on the console talking about ALSA. Mine never did when it was going wrong, the program would just hang if and only if called with the alsa devices. I looked at the logs and it didn't seem to have anything in it on the failures (I thought I'd posted them - my bad)

I've got an old SD image from about the time i found it. If I get the chance and remember i'll do some more testing - life is rather chaotic here atm.

**Comment by matthijskooijman** [comment]
> Unlike your error #745 which gives a line on the console talking about ALSA. Mine never did when it was going wrong, the program would just hang if and only if called with the alsa devices. I looked at the logs and it didn't seem to have anything in it on the failures (I thought I'd posted them - my bad)

Well, the error I'm seeing is from libalsa itself, any error messages from omxplayer ended up only in the logs. So my hypothesis was that in your case, the opening of the device would work, but some subsequent configuration of the device would fail and libalsa would perhaps not print anything about that. However, the error message from omxplayer would have shown up in the `--genlog` output, so if you looked at that, maybe it is completely unrelated after all.

In any case, if you can't reproduce, maybe it was fixed in the meanwhile somehow, or maybe some alsa config change (I never really understand those config files...).

**Comment by Foldi-One** [comment]
Yeah, ALSA takes me a few eons of reading and thinking to set my mind into the way it works (to be promptly half forgotten the moment I'm not needing it). I suspect in my case having device defined in local and not system level was the issue. As that is how it seems to be set up now and it does work..  So far I've not found the time to really dig back into this but skimming around the right files it seems to be the case. 

You could be correct they were connected though. Till I get round to reproducing it (if it still can be) its hard to say.
